### PR TITLE
Uses isinstance instead of issubclass in resampling strategy

### PR DESCRIPTION
**Pull request by franchuterivera**
Addresses #924

### CODE REVIEW

File: test/test_evaluation/test_train_evaluator.py [hunk 1]
```diff
@@ -1325,9 +1325,9 @@ def test_get_splitter_cv_object(self, te_mock):
         D.data['Y_train'] = np.array([0.0, 0.1, 0.2, 0.3, 0.4, 0.5])
         D.info['task'] = REGRESSION
         evaluator = TrainEvaluator()
-        evaluator.resampling_strategy = KFold
-        evaluator.resampling_strategy_args = {'folds': 4, 'shuffle': True,
+        evaluator.resampling_strategy_args = {'n_splits': 4, 'shuffle': True,
                                               'random_state': 5}
+        evaluator.resampling_strategy = KFold(**evaluator.resampling_strategy_args)
```
Comment (mfeurer) [inline test/test_evaluation/test_train_evaluator.py:1330]: Is the variable `evaluator.resampling_strategy_args` actually used anywhere now?

Comment (mfeurer) [reply test/test_evaluation/test_train_evaluator.py:1330]: or could these simply be argument to the constructor?

File: autosklearn/evaluation/train_evaluator.py [hunk 2]
```diff
@@ -979,10 +944,6 @@ def get_splitter(self, D: AbstractDataManager) -> Union[BaseCrossValidator, _Rep
                    D.info['task'] != MULTIOUTPUT_REGRESSION):
 
                     y = y.ravel()
-                if class_name == 'PredefinedSplit':
```
Comment (mfeurer) [inline autosklearn/evaluation/train_evaluator.py:982]: Ha, this issue was badly named. I like where it is going, but I was actually hoping to remove the checks `if class_name == 'XXX'`. Do you think that's possible?

Comment (franchuterivera) [reply autosklearn/evaluation/train_evaluator.py:982]: The only reason why I still kept them is because of the group parameter requirement of some splitters.

scikit-learn will raise a value error like this:
https://github.com/scikit-learn/scikit-learn/blob/2beed55847ee70d363bdbfe14ee4401438fba057/sklearn/model_selection/_split.py#L969

I think we can add a check to the automl fit, so that we don't even start if groups is missing. To do this, we call the `get_n_splits(groups=groups) ` method. makes sense?

File: test/test_evaluation/test_train_evaluator.py (hunk 1 above)
Comment (franchuterivera) [reply test/test_evaluation/test_train_evaluator.py:1330]: I think this will be a lot of arguments for the constructor. For example:

- shuffle or not
- folds
- groups
- number of repeats
- any new thing that fold split supports

This argument gives us flexibility I think. However if you think otherwise I can remove them and make them arguments, but then this will have to be passed to many classes like smbo, automl, estimators and so on.

Comment (mfeurer) [reply test/test_evaluation/test_train_evaluator.py:1330]: Oh, okay, in general yes. But in these tests it is basically not necessary, right?

File: autosklearn/evaluation/train_evaluator.py (hunk 2 above)
Comment (mfeurer) [reply autosklearn/evaluation/train_evaluator.py:982]: That's a new scikit-learn method which didn't exist back then and seems to solve our issues here :)

Yes, could you please move to using that?

Comment (franchuterivera) [reply autosklearn/evaluation/train_evaluator.py:1048]: I thought this resolved our problem. In the case of groupkfold it does not ask for a group argument, I have to do a `next()` on the splitter. I proposed a change with this but this maybe complicates the code a little? 

Let me know your thoughts...

File: autosklearn/automl.py [hunk 3]
```diff
@@ -832,6 +815,63 @@ def _fit_cleanup(self):
...
+            raise ValueError('Illegal resampling strategy: %s' % self._resampling_strategy)
+
+        if is_split_object:
+            TrainEvaluator.check_splitter_resampling_strategy(
+                X=X, y=y, task=task,
+                groups=self._resampling_strategy_arguments.get('groups', None),
+                resampling_strategy=self._resampling_strategy,
+            )
+
+        if self._resampling_strategy in [
```
Comment (mfeurer) [inline autosklearn/automl.py:858]: Nitpick, but wouldn't an `elif` be nicer here?

File: autosklearn/automl.py [hunk 4]
```diff
@@ -832,6 +815,63 @@ def _fit_cleanup(self):
...
+            'partial-cv',
+            'partial-cv-iterative-fit',
+        ] and self._ensemble_size != 0:
+            raise ValueError("Resampling strategy %s cannot be used "
+                             "together with ensembles." % self._resampling_strategy)
+
+        if self._resampling_strategy in ['partial-cv',
```
Comment (mfeurer) [inline autosklearn/automl.py:865]: and here?

File: autosklearn/evaluation/train_evaluator.py (hunk 2 above)
Comment (mfeurer) [reply autosklearn/evaluation/train_evaluator.py:1048]: Would this be an issue to raise in the scikit-learn issue tracker? The change looks fine and is an overall improvement!

**Review by mfeurer — APPROVED** [review]

**Comment by mfeurer** [comment]
Not sure if it is worth updating this PR with my two nitpicks, I guess not, so you can just merge if you're happy with it.

**Review by mfeurer — APPROVED** [review]
